res/controllers/Numark-NS6II-scripts.js

```diff
@@ -0,0 +1,1479 @@
+    });
+
+    this.deckLoadedConnection = engine.makeConnection(deck, "track_loaded", function(value) {
+        deckInfoCache.trackLoaded = value;
+    });
+    this.deckLoadedConnection.trigger();
+
+};
+
+NS6II.Display.prototype = new components.ComponentContainer();
+
+NS6II.PadMode = function(channelOffset) {
+    components.ComponentContainer.call(this, {});
+
+    this.constructPads = constructPad => {
+        this.pads = this.pads.map((_, padIndex) => constructPad(padIndex));
+    };
+    const makeParameterPressHandler = (control, onButtonDown) =>
+        new components.Button({
+            midi: [0x90 + channelOffset, control],
+            // never outconnect, as these buttons don't have LEDs
+            outConnect: false,
+            input: function(channelmidi, control, value, status, group) {
+                if (this.isPress(channelmidi, control, value, status)) {
+                    onButtonDown(channelmidi, control, value, status, group);
+                }
+            },
+        });
+    this.assignParameterPressHandlerLeft = onButtonDown => {
+        this.parameterLeft = makeParameterPressHandler(0x28, onButtonDown);
+    };
+    this.assignParameterPressHandlerRight = onButtonDown => {
+        this.parameterRight = makeParameterPressHandler(0x29, onButtonDown);
+    };
+    // this is a workaround for components, forEachComponent only iterates
+    // over ownProperties, so these have to constructed by the constructor here
+    // instead of being merged by the ComponentContainer constructor
+    this.pads = Array(8).fill(undefined);
+    const doNothing = () => {};
+    this.assignParameterPressHandlerLeft(doNothing);
+    this.assignParameterPressHandlerLeft(doNothing);
+};
+
+NS6II.PadMode.prototype = new components.ComponentContainer();
+
+NS6II.Pad = function(options) {
+    components.Button.call(this, options);
+};
+NS6II.Pad.prototype = new components.Button({
+    // grey could be an alternative as well as a backlight color.
+    off: NS6II.USE_BUTTON_BACKLIGHT ? NS6II.PAD_COLORS.RED.DIMMER : NS6II.PAD_COLORS.OFF,
+    outConnect: false,
+    sendShifted: true,
+    shiftControl: true,
+    shiftOffset: 8,
+});
+
+NS6II.PadModeContainers = {};
+
+NS6II.PadModeContainers.HotcuesRegular = function(channelOffset, hotCueOffset) {
+
+    NS6II.PadMode.call(this, channelOffset);
+
+    this.constructPads(i =>
+        new components.HotcueButton({
+            midi: [0x90 + channelOffset, 0x14 + i],
+            // shift: [0x94+channelOffset,0x1b+i],
+            number: i + 1 + hotCueOffset,
+            colorMapper: NS6II.padColorMapper,
+            // sendRGB: function(colorObj) {
+            //     this.send(NS6II.mixxxColorToDeviceColorCode(colorObj));
+            // },
+            off: NS6II.PAD_COLORS.OFF,
+        })
+    );
+    this.parameterLeft = new components.Button({
+        midi: [0x90, 0x28],
+        unshift: function() {
+            // TODO change hotcue page
```

Comment:
Is this outstanding? Or should it be removed?

Reply:
nah, its technically outstanding, but I really didn't want to bother implementing that because `components.HotcueButton` is not setup for that.